### PR TITLE
feat(stats): redesign dashboard UX

### PR DESCRIPTION
**Pull request by dean0x**
## Summary

- **Fix weighted savings %**: compute `tokens_saved / raw_tokens * 100` in rendering layer instead of SQL `AVG(savings_pct)` (80.5% vs 34.6%)
- **Simplify summary**: remove daily trend/sparkline, show Invocations + Raw tokens + Tokens saved (inline %) + 50-char bar
- **Rename "By Command" → "By Category"** with column headers; add new "By Command" section showing actual original commands (top 15 by savings)
- **Column headers** in all table sections (Category, Language, Mode, Command)
- **Extra spacing** after all section headers for visual clarity
- **`--verbose` flag** gates Parse Quality section (was always shown)
- **Cost estimates always shown** (previously behind `--cost` flag; flag kept as silent no-op for backward compat)
- **JSON output**: adds `weighted_savings_pct`, `by_original_cmd` array, always includes `cost_estimate`

## Test plan

- [x] `cargo test` — 2,476 tests passing, 0 failures
- [x] `cargo clippy -- -D warnings` — clean
- [x] `cargo fmt -- --check` — clean
- [x] `skim stats` — visual verification of new layout
- [x] `skim stats --verbose` — Parse Quality appears
- [x] `skim stats --format json` — all new fields present
- [x] `skim stats --help` — shows `--verbose`, no `--cost`
- [x] Scenario-based QA (10/10 scenarios passed)

### CODE REVIEW

File: crates/rskim/src/cmd/stats.rs [hunk 1]
```diff
@@ -5,14 +5,13 @@
 //! JSON output (`--format json`), cost estimates (`--cost`), and data clearing
 //! (`--clear`).
 
-use std::collections::HashMap;
 use std::io::{self, Write};
 use std::process::ExitCode;
 use std::time::UNIX_EPOCH;
 
 use colored::{ColoredString, Colorize};
```
Comment (dean0x) [inline crates/rskim/src/cmd/stats.rs:13]: **Documentation drift**: Module doc comment still references `--cost` flag which has been removed.

The doc says: "cost estimates (`--cost`)"

But this PR replaces it with always-on cost display + `--verbose` for parse quality.

**Fix**: Update to:
```rust
//! Queries the analytics SQLite database and displays a summary of token
//! savings across all skim invocations. Supports time filtering (`--since`),
//! JSON output (`--format json`), verbose parse quality (`--verbose`),
//! and data clearing (`--clear`). Cost estimates are always shown.
```

File: crates/rskim/src/cmd/stats.rs [hunk 2]
```diff
@@ -155,30 +149,44 @@ fn run_json(
...
+    let root = serde_json::json!({
+        "summary": {
+            "invocations": summary.invocations,
+            "raw_tokens": summary.raw_tokens,
+            "compressed_tokens": summary.compressed_tokens,
+            "tokens_saved": summary.tokens_saved,
+            "avg_savings_pct": summary.avg_savings_pct,
+            "weighted_savings_pct": weighted_savings_pct,
+        },
         "daily": daily,
```
Comment (dean0x) [inline crates/rskim/src/cmd/stats.rs:179]: **DRY violation: Duplicated computation at line 309**

The `weighted_savings_pct` formula is duplicated:
- Line 154-158 (here in `run_json`)
- Line 309-313 in `render_summary`

If the formula needs to change (e.g., rounding, clamping), both sites must be updated in lockstep.

**Fix**: Extract a shared function:
```rust
fn weighted_savings_pct(summary: &AnalyticsSummary) -> f64 {
    if summary.raw_tokens > 0 {
        (summary.tokens_saved as f64 / summary.raw_tokens as f64) * 100.0
    } else {
        0.0
    }
}
```
Then call `weighted_savings_pct(&summary)` in both places.

File: crates/rskim/src/cmd/stats.rs [hunk 3]
```diff
@@ -408,66 +306,59 @@ fn render_summary(
...
+        (summary.tokens_saved as f64 / summary.raw_tokens as f64) * 100.0
+    } else {
+        0.0
+    };
+
     writeln!(w, "{}", section_header("Summary"))?;
+    writeln!(w)?;
     writeln!(
         w,
-        "  Invocations:    {}",
```
Comment (dean0x) [inline crates/rskim/src/cmd/stats.rs:414]: **Inconsistent rendering**: Column header (5 cols) does not match data row format (6 cols with trailing bar).

Header at line 355:
```
CATEGORY | CALLS | SAVED | REDUCTION | AVG TIME
```

But data at line 361 appends a 6th value: `render_bar(cmd.avg_savings_pct, BAR_WIDTH)`

Other sections (`render_by_language`, `render_by_mode`) do the same. But the new `render_by_original_cmd` omits the bar entirely, creating visual inconsistency across the dashboard.

**Fix**: Add the bar to `render_by_original_cmd` to match the pattern, or document the intentional omission.

File: crates/rskim/src/analytics/mod.rs [hunk 4]
```diff
@@ -486,6 +503,34 @@ impl AnalyticsDb {
...
+    ) -> anyhow::Result<Vec<OriginalCommandStats>> {
+        let (where_clause, params) = since_clause(since);
+        let sql = format!(
+            "SELECT original_cmd, COUNT(*) as cnt, \
+             SUM(CASE WHEN raw_tokens > compressed_tokens THEN raw_tokens - compressed_tokens ELSE 0 END) as saved, \
+             AVG(savings_pct) as avg_pct, AVG(duration_ms) as avg_dur \
+             FROM token_savings {where_clause} \
+             GROUP BY original_cmd \
+             ORDER BY saved DESC \
+             LIMIT 15"
```
Comment (dean0x) [inline crates/rskim/src/analytics/mod.rs:534]: **SQL calculation style inconsistency**: Two different calculation approaches for tokens saved.

Existing `query_by_command` (line 424): `SUM(raw_tokens - compressed_tokens)` — computes with COALESCE, clamps in Rust.

New `query_by_original_cmd` (here): `SUM(CASE WHEN raw_tokens > compressed_tokens THEN ... ELSE 0 END)` — clamps in SQL.

Both produce correct results, but the inconsistency makes code harder to maintain.

**Fix**: Unify on the SQL-level CASE pattern across all query methods for consistency.

File: crates/rskim/src/cmd/stats.rs [hunk 5]
```diff
@@ -511,10 +408,16 @@ fn render_by_mode(
...
+    writeln!(
+        w,
+        "  {:<COL_NAME$}  {:>COL_COUNT$}  {:>COL_SAVED$}  {:<9}",
+        "MODE", "FILES", "SAVED", "REDUCTION"
+    )?;
     for mode in by_mode {
         writeln!(
             w,
-            "  {:<COL_NAME$} {:>COL_COUNT$} files  {:>COL_SAVED$} saved  {}  {}",
+            "  {:<COL_NAME$}  {:>COL_COUNT$}  {:>COL_SAVED$}  {}  {}",
```
Comment (dean0x) [inline crates/rskim/src/cmd/stats.rs:432]: **Missing edge-case test**: `truncate_cmd_display` has boundary condition for very small `max_chars` values.

The function uses `saturating_sub(3)` to reserve space for `"..."`. When `max_chars` is 0, 1, or 2:
- `end` becomes 0
- Result would be `"..."` (3 chars), exceeding the stated contract

While `DISPLAY_CMD_LEN` is hardcoded to 30 today, the function is a general-purpose helper with no documented precondition.

**Fix**: Add boundary-case tests or document: `/// max_chars must be >= 3`

File: crates/rskim/src/cmd/stats.rs [hunk 6]
```diff
@@ -549,11 +500,6 @@ fn render_cost_section(
 ) -> anyhow::Result<()> {
     let pricing = PricingModel::from_cost_override(cost_override);
     writeln!(w, "{}", section_header("Cost Estimates"))?;
-    writeln!(
-        w,
-        "  Rate:      ${:.2}/MTok ({})",
```
Comment (dean0x) [inline crates/rskim/src/cmd/stats.rs:554]: **Missing test**: `render_by_original_cmd` has early-return for empty input (line 453) that is not directly tested.

Since this is entirely new code, explicit coverage is recommended.

**Fix**: Add a unit test:
```rust
#[test]
fn test_render_by_original_cmd_empty() {
    let mut buf = Vec::new();
    render_by_original_cmd(&mut buf, &[]).unwrap();
    assert!(buf.is_empty());
}
```

**Comment by dean0x** [comment]
## Code Review Summary: stats-dashboard-redesign

**Overall Recommendation**: CHANGES_REQUESTED (6 blocking issues identified)

### Blocking Issues
All have ≥80% confidence and require resolution before merge:

1. **Documentation drift** (stats.rs:5) — Module doc comment still references removed `--cost` flag
2. **DRY violation** (stats.rs:154) — Duplicated `weighted_savings_pct` formula (also at line 309)
3. **Inconsistent rendering** (stats.rs:355) — Column headers don't match data row format (header: 5 cols, data: 6 with bar)
4. **SQL calculation inconsistency** (analytics/mod.rs:514) — Two different approaches to computing tokens_saved (SQL CASE vs Rust COALESCE)
5. **Missing edge-case test** (stats.rs:440) — `truncate_cmd_display` boundary condition for max_chars < 3
6. **Missing empty-input test** (stats.rs:463) — `render_by_original_cmd` early-return branch untested

### Lower-Confidence Suggestions (60-79%)
These are optional but recommended:

- Missing progress bar in `render_by_original_cmd` (65%) — Other sections include trailing bar visualization
- Missing `--verbose` flag parsing test (80% should-fix) — New flag needs end-to-end CLI integration test
- Test consolidation: `test_run_json_with_cost` is now near-duplicate of `test_run_json_with_data` (85% should-fix)

### What's Working Well

✅ **Excellent refactor fundamentals:**
- Weighted % computation correctly moved to rendering layer (80.5% vs 34.6%)
- Clean trait extension pattern on `AnalyticsStore`
- New `render_by_original_cmd` follows established function pattern
- Good deletions (`render_sparkline`, calendar math) — 130+ lines of complexity removed
- Comprehensive test coverage (73 tests passing)

✅ **Good architectural separation:**
- Data access (AnalyticsDb), presentation (render_* functions), orchestration (run_dashboard/run_json) all cleanly separated
- No layering violations

### Inline Comments Posted
See the 6 discussion threads above for detailed fixes and code samples.

---

**Reviewer Notes**: Confidence levels based on analysis from 8 specialized reviewers (architecture, complexity, consistency, regression, testing, rust, security, performance). Inline comments contain specific code samples where applicable. Issues marked BLOCKING are consensus findings (90%+ or multiple reviewers 80%+).

*Generated by: git-agent (devflow:git)*
